File: 2025/go/day02/part2.go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

func main() {
	lines, err := ReadLines(FileName)

	if err != nil {
		panic(err)
	}

	var invalidIds []int

	for i, _ := range lines {
		lNum, err := strconv.Atoi(strings.Split(lines[i], "-")[0])
		if err != nil {
			panic("We cannot parse Left Index")
		}
		rNum, err := strconv.Atoi(strings.Split(lines[i], "-")[1])

		if err != nil {
			panic("We cannot parse Right Index")
		}

		for n := lNum; n <= rNum; n++ {
			if isInvalid(n) {
				invalidIds = append(invalidIds, n)
			}
		}
	}

	var sum int
	for _, value := range invalidIds {
		sum += value
	}

	fmt.Println(sum)
}

func isInvalid(num int) bool {
	s := strconv.Itoa(num)
	l := len(s)

	for b := 1; b <= l / 2; b++ {
		if l % b != 0 {
			continue
		}

		block := s[0:b]
		count := l / b
		repeated := strings.Repeat(block, count)

		if repeated == s {
			return true
		}
	}	

	return false
}

File: 2025/js/day02/part2.mjs
import fs from 'fs';

const fileName = 'part2.txt';

const contents = fs.readFileSync(fileName, {
  encoding: 'utf8',
});

const eachNumArr = contents.replaceAll('\n', '').split(',');

const invalidIds = [];

function isInvalid(num) {
  const s = num.toString();
  const L = s.length;

  for (let b = 1; b <= L / 2; b++) {
    if (L % b !== 0) continue;

    const block = s.slice(0, b);
    const repeated = block.repeat(L / b);

    if (repeated === s) return true;
  }

  return false;
}

for (let k = 0; k < eachNumArr.length; k++) {
  const ids = eachNumArr[k].split('-');
  const lNum = +ids[0];
  const rNum = +ids[1];

  for (let n = lNum; n <= rNum; n++) {
    if (isInvalid(n)) {
      invalidIds.push(n);
    }
  }
}

const sum = invalidIds.reduce((acc, num) => acc + num, 0);
console.log(sum);
